services/powchain/src/main/java/tech/pegasys/teku/services/powchain/PowchainService.java

```diff
@@ -66,29 +67,50 @@ public class PowchainService extends Service {
   private final List<Web3j> web3js;
   private final OkHttpClient okHttpClient;
 
-  public PowchainService(final ServiceConfig serviceConfig, final PowchainConfiguration powConfig) {
-    checkArgument(powConfig.isEnabled());
+  public PowchainService(
+      final ServiceConfig serviceConfig,
+      final PowchainConfiguration powConfig,
+      final Optional<Web3j> trustedWeb3j,
+      final Optional<String> trustedEndpoint) {
```

Comment:
It seems weird to be passing these in as separate things.  Would probably be cleaner to just have a `Web3jProvider` interface that we pass the appropriate implementation in for - either getting it from the execution engine provider or creating new Web3jEth1Provider instances.  So we'd basically be moving the `if (trustedWeb3j.isPresent())` if and else conditions to be the two separate implementations of this interface rather than being embedded in here.